scripts/h5toobj.py

```diff
@@ -290,42 +225,45 @@ def dask_from_chunks(chunks, out_name, base_name):
 
     use_rados = args.ceph_pool is not None
     if use_rados:
-        store = open_rados(args.ceph_conf, args.ceph_pool)
+        obj_store = RadosChunkStore(args.ceph_conf, args.ceph_pool,
+                                    args.ceph_keyring)
+        pool_stats = obj_store.ioctx.get_stats()
+        logger.info("Connected to pool %s. Currently holds %d objects "
+                    "totalling %g GB", args.ceph_pool,
+                    pool_stats['num_objects'], pool_stats['num_bytes'] / 2**30)
         ts_pbs.add("ceph_pool", args.ceph_pool, immutable=True)
         with open(args.ceph_conf, "r") as ceph_conf:
             ts_pbs.add("ceph_conf", ceph_conf.readlines(), immutable=True)
-        save = functools.partial(write_chunk_rados, store)
     else:
-        store = open_s3(args.s3_url)
+        obj_store = S3ChunkStore(args.s3_url)
         ts_pbs.add("s3_endpoint", args.s3_url, immutable=True)
-        save = functools.partial(write_chunk_s3, store)
-    # Convert HDF5 visibilities to complex64 on the fly
-    vis_save = lambda obj, x: save(obj, x.view(np.complex64)[..., 0])
 
     target_object_size = args.obj_size * 2 ** 20
     dask_graph = {}
     schedule = dask.threaded.get
     output_keys = []
-    for dataset in ('timestamps', 'correlator_data', 'weights',
-                    'weights_channel', 'flags'):
-        arr = h5_file['Data'][dataset]
-        dask_graph[dataset] = arr
-        base_name = '/'.join((args.base_name, program_block, stream, dataset))
-        shape = (min(arr.shape[0], max_dumps),) + arr.shape[1:]
+    h5_store = DictOfArraysChunkStore(**h5_file['Data'])
+    for dataset in h5_store.arrays:
+        dataset = str(dataset)
+        arr = h5_store.arrays[dataset]
         dtype = arr.dtype
+        shape = arr.shape
         if dataset == 'correlator_data':
-            shape = shape[:-1]
             dtype = np.dtype(np.complex64)
+            shape = shape[:-1]
+        copy_dataset = 'copy_' + dataset
+        dask_graph[copy_dataset] = arr
```

Comment:
Does this actually do anything? I don't think this key is referenced anywhere else in the graph.

Reply:
It does. The keys spat out by `dsk_from_chunks` in the construction of `dsk` refer to it. Otherwise there is no reference to the actual data in the graph.

Reply:
Removed.